src/main/java/org/openstreetmap/josm/plugins/pt_assistant/gui/PTAssistantSnackbar.java

```diff
@@ -0,0 +1,94 @@
+// License: GPL. For details, see LICENSE file.
+
+package org.openstreetmap.josm.plugins.pt_assistant.gui;
+
+import static org.openstreetmap.josm.tools.I18n.tr;
+
+import java.awt.Color;
+import java.awt.Cursor;
+import java.awt.Font;
+import java.awt.GridBagLayout;
+import java.util.Timer;
+import java.util.TimerTask;
+import javax.swing.JButton;
+import javax.swing.JPanel;
+import javax.swing.border.CompoundBorder;
+import javax.swing.border.EmptyBorder;
+import javax.swing.border.EtchedBorder;
+
+import org.openstreetmap.josm.gui.MainApplication;
+import org.openstreetmap.josm.gui.widgets.JMultilineLabel;
+import org.openstreetmap.josm.tools.Logging;
+import org.openstreetmap.josm.tools.GBC;
+import org.openstreetmap.josm.gui.MapFrame;
+import org.openstreetmap.josm.tools.ImageProvider;
+
+/**
+ * Snackbars (or useful messages) that are displayed on the bottom of the map
+ *
+ * @author sudhanshu2
+ */
+public class PTAssistantSnackbar extends JPanel {
```

Comment:
@sudhanshu2 Looks pretty good to me. Just a few suggestions:

The JOSM API around `addTopPanel()` and `removeTopPanel()` doesn't really support this use case of multiple notifications that disappear after a few seconds, as you already discovered.

I think this kind of messages is intended for messages that appear and stay until the user closes them. Like some kind of warning (JOSM core warns about potentially misaligned imagery), or I could imagine you can use it for some kind of tutorial or guide that explains how to use the plugin. Or even some kind of simple wizard that can accept inputs.

Implementation-wise, you could use the singleton pattern for the Snackbar/TopPanel, so always only one instance of a class is added as TopPanel.
For notifications that only stay a few seconds, [`Notification`](https://github.com/openstreetmap/josm/blob/master/src/org/openstreetmap/josm/gui/Notification.java)s from JOSM core are useful. These already have the timing stuff built-in and will auto-close after a set amount of time, with correct handling of duplicates.